src/file/file.rpc.ts

```diff
@@ -41,30 +48,45 @@ export class FileServer {
     // Register the health service
     this.grpcHealthCheck = new GrpcHealthCheck(healthCheckStatusMap);
     this.server.addService(HealthService, this.grpcHealthCheck);
+    this.server.addService(
+      file_proto.FileService.service,
+      {
+        GenerateKey: this.wrapper(this.generateKey),
+        CreateUpload: this.wrapper(this.createUpload),
+        UpdateUploadID: this.wrapper(this.updateUpload),
+        GetUploadByID: this.wrapper(this.getUploadByID),
+        DeleteUploadByID: this.wrapper(this.deleteUploadByID),
+        GetFileByID: this.wrapper(this.getFileByID),
+        GetFileByKey: this.wrapper(this.getFileByKey),
+        GetFilesByFolder: this.wrapper(this.getFilesByFolder),
+        CreateFile: this.wrapper(this.createFile),
+        DeleteFile: this.wrapper(this.deleteFile),
+        IsAllowed: this.wrapper(this.isAllowed),
+      }
+    );
+    this.server.bind(`0.0.0.0:${port}`, grpc.ServerCredentials.createInsecure());
+  }
 
-    this.server.addService(file_proto.FileService.service, {
-      GenerateKey: this.generateKey,
-      CreateUpload: this.createUpload,
-      UpdateUploadID: this.updateUpload,
-      GetUploadByID: this.getUploadByID,
-      DeleteUploadByID: this.deleteUploadByID,
-      GetFileByID: this.getFileByID,
-      GetFileByKey: this.getFileByKey,
-      GetFilesByFolder: this.getFilesByFolder,
-      CreateFile: this.createFile,
-      DeleteFile: this.deleteFile,
-      IsAllowed: this.isAllowed,
-    });
+  private wrapper (func: any) : any {
+    return async (call:any, callback:any) => {
+      try {
+        apm.startTransaction(func.name, 'monitoringFS');
+        const res = await func(call, callback);
+        apm.endTransaction('successful');
+        callback(null, res);
+      } catch (err) {
+        apm.endTransaction('failed');
```

Comment:
it seems there is no good way to easily translate grpc ststus codes into their corresponding name (i.e 0:'OK',...).
looks like its because of the way ts is transpiled into js with imported enums, but I'm not sure.
so I created a function for doing that translation. it's quite ugly, but it works. (utils/errors/grpc.status.ts).
If you have a better solution, let me know.

Reply:
@Shahar-Y There's `grpc.status[err.code]`, you don't need a function for it, have fun.